deploy/terraform-aws/main.tf

```diff
@@ -293,6 +287,113 @@ resource "aws_iam_role_policy" "k8s_node" {
 EOF
 }
 
+resource "aws_iam_role_policy_attachment" "k8s-policy-attachment" {
+  policy_arn = aws_iam_policy.k8s_node_k8s_policy.arn
+  role = aws_iam_role.k8s_node.name
+}
+
+resource "aws_iam_policy" "k8s_node_kip_policy" {
+  name = "k8s-node-${var.cluster_name}-kip"
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:AssociateIamInstanceProfile",
+        "ec2:AuthorizeSecurityGroupIngress",
+        "ec2:CreateRoute",
+        "ec2:CreateSecurityGroup",
+        "ec2:CreateTags",
+        "ec2:DeleteRoute",
+        "ec2:DeleteSecurityGroup",
+        "ec2:DescribeAvailabilityZones",
+        "ec2:DescribeDhcpOptions",
+        "ec2:DescribeImages",
+        "ec2:DescribeInstances",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DescribeRouteTables",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVolumes",
+        "ec2:DescribeVpcs",
+        "ec2:ModifyInstanceAttribute",
+        "ec2:ModifyInstanceCreditSpecification",
+        "ec2:ModifyVolume",
+        "ec2:RevokeSecurityGroupIngress",
+        "ec2:RunInstances",
+        "ec2:TerminateInstances",
+        "ecr:BatchGetImage",
+        "ecr:GetAuthorizationToken",
+        "ecr:GetDownloadUrlForLayer",
+        "iam:PassRole"
+      ],
+      "Resource": [
+        "*"
+      ]
+    }
+  ]
+}
+EOF
+}
+
+
+resource "aws_iam_role_policy_attachment" "kip-policy-attachment" {
+  policy_arn = aws_iam_policy.k8s_node_kip_policy.arn
+  role = aws_iam_role.k8s_node.name
+}
+
+resource "aws_iam_policy" "k8s_node_ssm_policy" {
```

Comment:
There's a managed IAM policy for SSM, we should use that one.

Reply:
good point! changed it to `arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore` arn